core/state/reader.go

```diff
@@ -48,6 +48,19 @@ type Reader interface {
 	// - The returned storage slot is safe to modify after the call
 	Storage(addr common.Address, slot common.Hash) (common.Hash, error)
 
+	// ContractCode returns the code associated with a particular account.
+	//
+	// - Returns an empty code if it does not exist
+	// - It can return an error to indicate code doesn't exist
```

Comment:
This breaks consistency to the rest of the interface. I have done this to avoid any semantics change in this PR.

I am a bit puzzled why the statedb is recording "not found" errors for code but not for storage or account.

Reply:
It says "empty code if it does not exist". Pls clarify.
It says "_can_ return an error to indicate code doesn't exist". Well, does it, or not?

Reply:
Sorry clarified. It will return an error if code is not found.